docs/contribute.md

```diff
@@ -0,0 +1,244 @@
+
+Install pre-commit either with your system package manager.
+Alternatively you can install pre-commit into your Python user.
+
+```shell
+python -m pip install --user pre-commit
+```
+
+Once installed, set up the git hook scripts to run on every commit.
+
+```shell
+pre-commit install
+```
+
+
+## Create development environment
+
+After satisfying the pre-requisites, you are ready to create your development environment.
+`plone.api` uses `tox` as a wrapper around `coredev.buildout` to simplify development, whereas Plone core uses `coredev.buildout` directly.
+
+Start by changing your working directory to your project folder, and download the latest `plone.api` source code.
+
+```shell
+cd <your_project_folder>
+git clone https://github.com/plone/plone.api.git
+```
+
+Next go into the newly created directory, and build your environment.
+
+```shell
+cd plone.api
+tox
+```
+
+Go make some tea while `tox` runs all tasks listed by issuing the command `tox -l`.
+
+Open `tox.ini` in your code editor to see all configured commands and what they do.
+Some helpful `tox` commands are shown below.
+
+```shell
+tox -e py39-plone-60  # run all tests for Python 3.9 and Plone 6
+tox -e plone6docs     # build documentation
+tox -e livehtml       # build, serve, and reload changes to documentation
+tox -l                # list all tox environments
+```
+
+
+(git-workflow)=
+
+## git
+
+Use the following git branches when contributing to `plone.api`.
+
+feature branches
+:   All development for a new feature or bug fix must be done on a new branch.
+
+`master`
+:   Pull requests should be made from a feature branch against the `master` branch.
+    When features and bug fixes are complete and approved, they are merged into the `master` branch.
+
+```{seealso}
+{ref}`plone:contributing-core-work-with-git-label`
+```
+
+
+## Continuous integration
+
+`plone.api` uses GitHub workflows for continuous integration.
+On every push to the `master` branch, GitHub runs its workflows for all tests and code quality checks.
+GitHub workflows are configured in the directory `.github/workflows` at the root of this package.
+
+
+## Documentation
+
+For every feature change or addition to `plone.api`, you should add documentation of it.
+`plone.api` uses [MyST](https://myst-parser.readthedocs.io/en/latest/) for documentation syntax.
+
+```{seealso}
+{doc}`plone:contributing/documentation/index`
+```
+
+After adding or modifying documentation, you can build the documentation with the following command.
+
+```shell
+tox -e plone6docs
+```
+
+Alternatively, you can automatically reload changes to the documentation as you edit it in a web browser.
+
+```shell
+tox -e livehtml
+```
+
+The [`plone.api` documentation](https://6.docs.plone.org/plone.api) is automatically generated from the documentation source files when its submodule is updated in the [main Plone `documentation` repository](https://github.com/plone/documentation/).
+
+
+### Add a function to an existing module
```

Comment:
```suggestion
### Add a method to an existing module
```

Reply:
Well, the whole plone.api is build with functions (not classes+methods), so it is explained below. So IMO functions is fine.

Reply:
The implementation of a feature is of course a function, not a classes method. That's why I wrote this header like this back then.
It's just that I realised that in the rest of the documentation it is often spoken of 'API methods'.
But I am fine with keeping the header like it is.